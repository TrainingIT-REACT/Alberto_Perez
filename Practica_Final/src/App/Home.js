import React, { Component } from 'react';
import { connect } from 'react-redux';

// Acciones
import { getSongs } from './actions/song';

//components
import SongList from './components/SongList'
import SearchSong from './SearchSong'

class Home extends Component {

  componentDidMount() {
    this.props.getSongs();
  }

  renderSongs() {
    if (this.props.isLoading) {
      return <p>Cargando...</p>
    } else if (this.props.error) {
      return <p>Hubo un error al obtener los datos :(</p>
    } else {
      //obtenemos 5 canciones al azar
      let index = Math.floor(Math.random()*this.props.song.list.length)
      //controlamos el indice inferior
      if(index < 5){
        index = index + 5
      }
      let list = this.props.song.list.slice(index-5,index)
      return <SongList songs={list} history={this.props.history}/>
    }
  }

  render() {
    return (
      <div>
        <h2>Canciones Recomendadas</h2>
        { this.renderSongs() }
        <SearchSong history={this.props.history}/>
      </div>
    );
  }
}

const mapStateToProps = (state) => ({
  ...state
});

const mapDispatchToProps = (dispatch) => ({
  getSongs: () => dispatch(getSongs())
})

export default connect(
  mapStateToProps,
  mapDispatchToProps
) (Home);