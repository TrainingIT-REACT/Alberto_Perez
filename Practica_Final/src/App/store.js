import { createStore, combineReducers, applyMiddleware } from "redux";
import thunk from 'redux-thunk';
//import promise from 'redux-promise-middleware';

// Reducers
import album from './reducers/album';
import song from './reducers/song';
import user from './reducers/user';

export default createStore(combineReducers({ user, album, song }), applyMiddleware(thunk));
