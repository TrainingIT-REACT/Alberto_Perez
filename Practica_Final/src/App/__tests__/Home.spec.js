import React from 'react';
import { render, mount } from 'enzyme';

import { Provider } from "react-redux";
import store from '../store';

// Componente
import Home from '../Home';



describe("Home", () => {
  describe("Render", () => {
    let wrapper;
    // Inicializamos el componente en un beforeEach para
    // evitar tener que repetir esta línea en cada test
    beforeEach(() => {
      wrapper = render(<Provider store={store}><Home/> </Provider>);
    });


    it('renders correctly', () => {
      expect(wrapper).toMatchSnapshot();
    });


    it('should add the HTML elements', () => {      
      // Comprobamos los distintos aspectos de HTML      
      expect(wrapper.find('h2')).toBeTruthy();
    }); 

  });

  /*
  describe("Features", () => {
    let wrapper;

    // Inicializamos el componente en un beforeEach para
    // evitar tener que repetir esta línea en cada test
    beforeEach(() => {
      wrapper = mount(<Provider store={store}><Home/> </Provider>);
    });

    it('should pass the ref to the input', () => {
      // En este caso necesitamos montar el componente ya que la nueva API de
      // referencias solo se instancian con mount
      const instance = wrapper.instance();
      instance.setProps({ isLoading: true });

      expect(instance.find('p')).toBeTruthy();
    });

  });
  */

});


