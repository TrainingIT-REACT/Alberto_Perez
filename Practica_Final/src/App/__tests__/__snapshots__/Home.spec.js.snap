// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Render renders correctly 1`] = `
Array [
  <div>
    <h2>
      Top 5 de Canciones Recomendadas
    </h2>
    <ul />
  </div>,
  " ",
]
`;
